crates/bevy_app/src/app.rs

```diff
@@ -608,10 +607,16 @@ impl App {
         self
     }
 
-    pub fn sub_app_mut(&mut self, label: impl SubAppLabel) -> Option<&mut App> {
-        let label = Box::new(label) as BoxedSubAppLabel;
+    /// Retrieves a "sub app" stored inside this [App]. This will panic if the sub app does not exist.
+    pub fn sub_app(&mut self, label: impl AppLabel) -> &mut App {
+        self.get_sub_app(label)
+            .expect("SubApp with the given label does not exist")
```

Comment:
This should return the label's identity in the error message.

Reply:
Good call!